test/org/sagebionetworks/bridge/play/controllers/FPHSControllerTest.java

```diff
@@ -0,0 +1,215 @@
+        String json = Helpers.contentAsString(result);
+        return BridgeObjectMapper.get().readTree(json);
+    }
+    
+    private void setExternalIdentifierPost(ExternalIdentifier externalId) throws Exception {
+        String json = BridgeObjectMapper.get().writeValueAsString(externalId);
+        Http.Context context = TestUtils.mockPlayContextWithJson(json);
+        Http.Context.current.set(context);
+    }
+    
+    private void setFPHSExternalIdentifiersPost(List<FPHSExternalIdentifier> list) throws Exception {
+        String json = BridgeObjectMapper.get().writeValueAsString(list);
+        Http.Context context = TestUtils.mockPlayContextWithJson(json);
+        Http.Context.current.set(context);
+    }
+    
+    private void setData() throws Exception {
+        FPHSExternalIdentifier id1 = FPHSExternalIdentifier.create("foo");
+        id1.setRegistered(true);
+        
+        FPHSExternalIdentifier id2 = FPHSExternalIdentifier.create("bar");
+        
+        List<FPHSExternalIdentifier> identifiers = Lists.newArrayList(id1, id2);
+        
+        when(fphsService.getExternalIdentifiers()).thenReturn(identifiers);
+    }
+    
+    private void setUserSession() {
+        user = mock(User.class);
+        when(user.getHealthCode()).thenReturn("BBB");
+        
+        UserSession session = mock(UserSession.class);
+        when(session.getUser()).thenReturn(user);
+        
+        doReturn(session).when(controller).getAuthenticatedAndConsentedSession();
+        doReturn(session).when(controller).getAuthenticatedSession();
+    }
+    
+    @Test
+    public void verifyOK() throws Exception {
+        Result result = controller.verifyExternalIdentifier("foo");
+        JsonNode node = resultToJson(result);
+        
+        // No session is required
+        verifyNoMoreInteractions(authenticationService);
+        
+        assertEquals("External identifier is valid.", node.get("message").asText());
+    }
+    
+    @Test
+    public void verifyFails() throws Exception {
+        when(fphsService.verifyExternalIdentifier(any())).thenThrow(new EntityNotFoundException(FPHSExternalIdentifier.class));
+        
+        try {
+            controller.verifyExternalIdentifier("foo");
+            fail("Should have thrown exception");
+        } catch(EntityNotFoundException e) {
+            assertEquals("ExternalIdentifier not found.", e.getMessage());
+        }
+    }
+    
+    @Test
+    public void verifyFailsWhenNull() throws Exception {
+        when(fphsService.verifyExternalIdentifier(any())).thenThrow(new InvalidEntityException("ExternalIdentifier cannot be blank, null or missing."));
+        
+        try {
+            controller.verifyExternalIdentifier(null);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertEquals("ExternalIdentifier cannot be blank, null or missing.", e.getMessage());
+        }
+    }
```

Comment:
I think these are separate. In one case the user has sent an invalid identifier, the controller should pass that along (and not catch the exception that results). If the user does not pass the parameter, the controller gets a null, and I do want to test that the controller does exactly the same thing, it passes it along and returns the exception.